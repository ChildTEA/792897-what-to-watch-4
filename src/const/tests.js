const fullMoviesDescriptions = [
  {
    id: `0001`,
    title: `Title 1`,
    director: `Director 1`,
    genre: `Genre 1`,
    poster: `awesome-poster.jpg`,
    runTime: `1h 0m`,
    release: `2000`,
    starring: [`Awesome Actor`, `Awesome Actress`],
    preview: `awesome-preview.jpg`,
    videoPreviewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: `0002`,
    title: `Title 2`,
    director: `Director 2`,
    genre: `Genre 2`,
    poster: `awesome-poster.jpg`,
    runTime: `1h 0m`,
    release: `2000`,
    starring: [`Awesome Actor`, `Awesome Actress`],
    preview: `awesome-preview.jpg`,
    videoPreviewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: `0003`,
    title: `Title 3`,
    director: `Director 3`,
    genre: `Genre 3`,
    poster: `awesome-poster.jpg`,
    runTime: `1h 0m`,
    release: `2000`,
    starring: [`Awesome Actor`, `Awesome Actress`],
    preview: `awesome-preview.jpg`,
    videoPreviewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: `0004`,
    title: `Title 4`,
    director: `Director 4`,
    genre: `Genre 4`,
    poster: `awesome-poster.jpg`,
    runTime: `1h 0m`,
    release: `2000`,
    starring: [`Awesome Actor`, `Awesome Actress`],
    preview: `awesome-preview.jpg`,
    videoPreviewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
];

const shortMoviesDescriptions = [
  {
    id: `0001`,
    title: `Title 1`,
    preview: `awesome-preview.jpg`,
    videoPreviewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: `0002`,
    title: `Title 2`,
    preview: `awesome-preview.jpg`,
    videoPreviewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: `0003`,
    title: `Title 3`,
    preview: `awesome-preview.jpg`,
    videoPreviewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: `0004`,
    title: `Title 4`,
    preview: `awesome-preview.jpg`,
    videoPreviewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
];

const shortMovieDescription = shortMoviesDescriptions[0];
const fullMoviesDescription = fullMoviesDescriptions[0];


export {
  fullMoviesDescriptions,
  fullMoviesDescription,
  shortMoviesDescriptions,
  shortMovieDescription
};
