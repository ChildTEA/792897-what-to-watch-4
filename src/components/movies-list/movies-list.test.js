import React from "react";
import renderer from "react-test-renderer";
import MoviesList from "./movies-list.jsx";
import {fullMoviesDescriptions} from "../../const/tests.js";


const movies = fullMoviesDescriptions;

describe(`<MoviesList />`, () => {
  it(`Should MoviesList render correctly`, () => {
    const tree = renderer
      .create((
        <MoviesList
          movies={movies}
          onCardClick={() => {}}
        />
      ), {
        createNodeMock: () => {
          return {};
        }
      })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
