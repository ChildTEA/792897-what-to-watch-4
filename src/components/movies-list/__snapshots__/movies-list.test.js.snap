// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoviesList /> Should MoviesList render correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        data-id="0001"
        href="movie-page.html"
        onClick={[Function]}
      >
        Title 1
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        data-id="0002"
        href="movie-page.html"
        onClick={[Function]}
      >
        Title 2
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        data-id="0003"
        href="movie-page.html"
        onClick={[Function]}
      >
        Title 3
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        data-id="0004"
        href="movie-page.html"
        onClick={[Function]}
      >
        Title 4
      </a>
    </h3>
  </article>
</div>
`;
